.github/workflows/tox.yaml

```diff
@@ -2,51 +2,108 @@ name: Run tests with Tox
 
 on:
   push:
-    branches: [main]
+    branches:
+      - main
```

Comment:
it's kinda pointless to run these integration tests on merges to `main`

Reply:
you, alright!  i learned it by [watching you](https://github.com/charmed-kubernetes/charm-flannel/commit/1d18603292b88b659cc666ad0f876bb77c94f02b#diff-95557d53bf91069e59d82b9d8fcfaf52ec84a762858983edd5ef3c9b3e4c8191R4-R6).

nice catch, let's just test on PRs to main.